mne/_fiff/reference.py

```diff
@@ -192,22 +195,18 @@ def _apply_reference(inst, ref_from, ref_to=None, forward=None, ch_type="auto"):
 
 def _apply_dict_reference(inst, ref_dict):
     """Apply a dict-based custom EEG referencing scheme."""
-    _check_before_dict_reference(inst, ref_dict)
+    # this converts all keys to channel indices and all values to arrays of ch. indices:
+    ref_dict = _check_before_dict_reference(inst, ref_dict)
 
     data = inst._data
     orig_data = data.copy()
-    if ref_dict:
-        for key, value in ref_dict.items():
-            if isinstance(value, str):
-                value = [value]  # pick_channels expects a list
-            ref_from = pick_channels(inst.ch_names, value, ordered=True)
-            ref_to = pick_channels(inst.ch_names, [key], ordered=True)
-            ref_data = orig_data[..., ref_from, :].mean(-2, keepdims=True)
-            data[..., ref_to, :] -= ref_data
+    for ref_to, ref_from in ref_dict.items():
+        ref_data = orig_data[..., ref_from, :].mean(-2, keepdims=True)
+        data[..., [ref_to], :] -= ref_data
 
     with inst.info._unlock():
         inst.info["custom_ref_applied"] = FIFF.FIFFV_MNE_CUSTOM_REF_ON
-    return inst, data
+    return inst, None
```

Comment:
for REST and proj-type references, we return `None` as the second return value, because it's not straightforward/sensible to provide the subtracted array. I think we should do the same here...  technically it should be possible to figure out the subtracted array, but it's complicated to do so in cases like `ref_channels=dict(Pz=["P1", "P2"], P1=["P3", "P5"])`